pieforproviders/pieforproviders
# Support XLSX and XLS files upload

# 💅🏼 What issue does this fix?

Fixes https://github.com/pieforproviders/pieforproviders/issues/84

- Supports uploading of XLS and XLSX files
- Updates the route to `/cases/import` as `/csv-import` no longer makes sense. And replaces references to `CSVImport` with `CasesImport`

Screenshots

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/7039523/90469564-c7f05780-e0de-11ea-867c-7cf5500dca81.png">

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/7039523/90471286-641c5d80-e0e3-11ea-9aa6-f5cfccb4733a.png">


## 🍂 Before You Submit


## 🛷 Deployment Considerations


## 🧵 Steps to set up locally


Run `cd client && yarn` to install the dependencies

## 🧳 Steps to test


1. Log in and go to `/cases/import`
2. Upload an XLS cases file with data. You should find the data displayed in a table
3. Repeat Step 3 for XLSX and CSV files

## 🕯 Caveats, concerns

The code is tightly coupled to the sample sheet structure provided in https://github.com/pieforproviders/pieforproviders/issues/144. The code skips the first 3 rows as they are filled with instructions, headers, and column descriptions. If the template changes, we'll have to update the code. If a user removes these rows or changes the column order, the code might not work as expected. I would recommend moving the instructions and the column description rows to the application if possible.

We might want to use [React Virtualized](https://github.com/bvaughn/react-virtualized) or [React Window](https://github.com/bvaughn/react-window) if we anticipate too many rows in the uploaded file.
